api/types/database.go

```diff
@@ -533,6 +536,24 @@ func (d *DatabaseV3) CheckAndSetDefaults() error {
 		if d.Spec.AWS.Region == "" {
 			d.Spec.AWS.Region = region
 		}
+	case awsutils.IsRedshiftServerlessEndpoint(d.Spec.URI):
+		details, err := awsutils.ParseRedshiftServerlessEndpoint(d.Spec.URI)
+		if err != nil {
+			logrus.WithError(err).Warnf("Failed to parse Redshift Serverless endpoint %v.", d.Spec.URI)
+			break
+		}
```

Comment:
> Is it possible for the endpoint to fail to parse but still be valid?
Yes, it could be valid in the future when something gets changed/added on the AWS side. For example, we return errors today for AWS China regions but both auto-discovered databases and static databases (where users manually input `workgroup` field) mostly like still work with the `break` here. So an error here means we don't know the format but it could be valid.

Also returning an error here would prevent database agent from starting up.